scylla/src/transport/iterator.rs

```diff
@@ -853,8 +853,8 @@ where
             let result = (self.fetcher)(paging_state).await?;
             let response = result.into_non_error_query_response()?;
             match response.response {
-                NonErrorResponse::Result(result::Result::Rows(mut rows)) => {
-                    paging_state = rows.paging_state.take();
+                NonErrorResponse::Result(result::Result::Rows(rows)) => {
+                    paging_state = rows.paging_state.clone();
                     let (proof, send_result) = self
```

Comment:
Is there a reason for this change? Why we can't `.take()` anymore? I fail to see how does it relate to this commit (`codewide: share ResultMetadata by an Arc `)

Reply:
This is a leftover from before I extracted paging state from ResultMetadata. I'll fix this.